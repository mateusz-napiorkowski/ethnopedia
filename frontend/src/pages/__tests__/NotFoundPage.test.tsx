import '@testing-library/jest-dom';
import { render } from '@testing-library/react'
import NotFoundPage from "../NotFoundPage"
import { MemoryRouter, Routes, Route } from "react-router-dom";
import userEvent from "@testing-library/user-event";

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => mockUseNavigate,
}));

const user = userEvent.setup()

const renderComponent = () => {
    return render(
        <MemoryRouter initialEntries={[`/`]}>
            <Routes>
                <Route path="/" element={<NotFoundPage/>}/>
            </Routes>  
        </MemoryRouter>
    );
};

describe("NotFoundPage tests", () => {
    beforeEach(() => {
        jest.clearAllMocks()
    });

    it("should render not found page", () => {           
        const {container} = renderComponent()
        
        expect(container).toMatchSnapshot()
    })

    it("should call useNavigate with path to home page when go back to home page button is clicked", async () => {           
        const {getByText} = renderComponent()
        const goToHomePageButton = getByText(/powrót na stronę główną/i)

        await user.click(goToHomePageButton)

        expect(mockUseNavigate).toHaveBeenCalledWith("/")
    })
})