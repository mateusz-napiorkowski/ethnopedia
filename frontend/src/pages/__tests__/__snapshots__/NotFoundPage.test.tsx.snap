// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage tests should render not found page 1`] = `
<div>
  <section
    class="bg-white dark:bg-gray-900"
  >
    <div
      class="py-8 px-4 mx-auto max-w-screen-xl lg:py-16 lg:px-6"
    >
      <div
        class="mx-auto max-w-screen-sm text-center"
      >
        <h1
          class="mb-4 text-7xl tracking-tight font-extrabold lg:text-9xl text-primary-600 dark:text-primary-500"
        >
          404
        </h1>
        <p
          class="mb-4 text-3xl tracking-tight font-bold text-gray-900 md:text-4xl dark:text-white"
        >
          Coś poszło nie tak.
        </p>
        <p
          class="mb-4 text-lg font-light text-gray-500 dark:text-gray-400"
        >
          Przepraszamy, nie można znaleźć tej strony.
        </p>
        <button
          class="inline-flex text-white bg-primary-600 hover:bg-primary-800 focus:ring-4 focus:outline-none
                    focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center bg-sky-500 border-none
                    dark:focus:ring-primary-900 font-semibold my-4"
        >
          Powrót na stronę główną
        </button>
      </div>
    </div>
  </section>
</div>
`;
